src/persist/src/indexed/mod.rs

```diff
@@ -939,6 +941,33 @@ impl<L: Log, B: Blob> Indexed<L, B> {
     }
 }
 
+impl<L: Log, B: Blob> Indexed<L, B> {
+    /// Returns a [Description] of the collection identified by `id_str`.
+    fn get_description(&mut self, id_str: &str, res: PFutureHandle<Description<u64>>) {
+        res.fill((|| {
+            // TODO: Is apply_batched_cmd OK to use here?
+            self.drain_pending()?;
+
+            self.apply_unbatched_cmd(|state, _, _| {
+                let registration = state.id_mapping.iter().find(|s| s.name == id_str);
+                match registration {
+                    Some(registration) => {
+                        let arrangement = state
+                            .arrangements
+                            .get(&registration.id)
+                            .expect("missing trace");
+                        let upper = arrangement.get_seal();
+                        let since = arrangement.since();
+                        let lower = Antichain::from_elem(u64::minimum());
```

Comment:
I'm personally divided on (though leaning slightly away from) using Description for this given that, for a collection, the lower is always the same. IIUC Descriptions are more for describing batches than collections. As a mild precedent here, Arrangement/TraceReader in differential dataflow doesn't have anything that returns a Description, instead breaking it into read_upper/get_logical_compaction.

That said, I also have a feeling that we might have some followup tweaks to this (like maybe returning a map of this info for all the active collections in persist so it only has to be called once at startup?) and I don't want to get in your way any more than I have to, so happy to merge as-is for now and figure this out later. In which case, I just ask that you add a TODO about thinking the return type of this through more

Reply:
I thought about this for a bit and i think i agree with Dan. It's hard for me to imagine a scenario where a nonzero `lower` on an arrangement is useful (the only example I came up with is -- maybe if we had a scenario where we only persisted things for the last N hours, it would be nice to somehow be able to increment `lower` to indicate that -- but im not even sure if thats actually a correct usage of `lower` and batches or not since the sequence of batches is meant to be logically append only)

Reply:
Thanks! I'll add a comment describing these thoughts but merge as is.